### ISSUE TITLE
Add Alarm Command

### ISSUE THREAD

**Issue by doudz** [fairecasoimeme/ZiGate#179]
I bought a HEIMAN siren http://www.heimantech.com/product/153.html
I paired it successfully to zigate but now I need some command to raise alarm and currently zigate doesn't provide any
The siren supports cluster 0x0009
![image](https://user-images.githubusercontent.com/1262548/54717803-531b9580-4b59-11e9-9f93-3cb2b8f53359.png)

**Comment by doudz** [comment]
I think we need to support eCLD_AlarmsSignalAlarm and eCLD_AlarmsClearAlarm

**Comment by ISO-B** [comment]
Can you post which clusters it supports? Does it support 0x0502? It seems more correct for this device than alert cluster.

**Comment by doudz** [comment]
Extracted from my zigate.json
```json
{
            "addr": "c008",
            "discovery": "auto-discovered",
            "endpoints": [
                {
                    "clusters": [
                        {
                            "attributes": [
                                {
                                    "attribute": 0,
                                    "name": "zcl_version",
                                    "value": null
                                },
                                {
                                    "attribute": 1,
                                    "name": "application_version",
                                    "value": null
                                },
                                {
                                    "attribute": 2,
                                    "name": "stack_version",
                                    "value": null
                                },
                                {
                                    "attribute": 3,
                                    "name": "hardware_version",
                                    "value": null
                                },
                                {
                                    "attribute": 4,
                                    "data": "Heiman",
                                    "name": "manufacturer",
                                    "value": "Heiman"
                                },
                                {
                                    "attribute": 5,
                                    "data": "WarningDevice",
                                    "name": "type",
                                    "type": "str",
                                    "value": "WarningDevice"
                                }
                            ],
                            "cluster": 0
                        },
                        {
                            "attributes": [
                                {
                                    "attribute": 32,
                                    "data": 40,
                                    "name": "battery_voltage",
                                    "type": "float",
                                    "value": 4.0
                                },
                                {
                                    "attribute": 33,
                                    "data": 192,
                                    "name": "battery_percent",
                                    "type": "int",
                                    "value": 192
                                },
                                {
                                    "attribute": 65533
                                }
                            ],
                            "cluster": 1
                        },
                        {
                            "attributes": [
                                {
                                    "attribute": 0
                                },
                                {
                                    "attribute": 65533
                                }
                            ],
                            "cluster": 3
                        },
                        {
                            "attributes": [
                                {
                                    "attribute": 0
                                },
                                {
                                    "attribute": 65533
                                }
                            ],
                            "cluster": 4
                        },
                        {
                            "attributes": [
                                {
                                    "attribute": 0,
                                    "data": "0000000000",
                                    "name": "zone_status",
                                    "value": {
                                        "ac_fault": false,
                                        "alarm1": false,
                                        "alarm2": false,
                                        "battery_defect": false,
                                        "low_battery": false,
                                        "restore": false,
                                        "supervision": false,
                                        "tamper": false,
                                        "test_mode": false,
                                        "trouble": false
                                    }
                                },
                                {
                                    "attribute": 1,
                                    "data": "0000000000",
                                    "name": "zone_status1",
                                    "value": {
                                        "ac_fault": false,
                                        "alarm1": false,
                                        "alarm2": false,
                                        "battery_defect": false,
                                        "low_battery": false,
                                        "restore": false,
                                        "supervision": false,
                                        "tamper": false,
                                        "test_mode": false,
                                        "trouble": false
                                    }
                                },
                                {
                                    "attribute": 2,
                                    "data": "0000000000",
                                    "name": "zone_status11",
                                    "value": {
                                        "ac_fault": false,
                                        "alarm1": false,
                                        "alarm2": false,
                                        "battery_defect": false,
                                        "low_battery": false,
                                        "restore": false,
                                        "supervision": false,
                                        "tamper": false,
                                        "test_mode": false,
                                        "trouble": false
                                    }
                                },
                                {
                                    "attribute": 16,
                                    "data": "0000000000",
                                    "name": "zone_status111",
                                    "value": {
                                        "ac_fault": false,
                                        "alarm1": false,
                                        "alarm2": false,
                                        "battery_defect": false,
                                        "low_battery": false,
                                        "restore": false,
                                        "supervision": false,
                                        "tamper": false,
                                        "test_mode": false,
                                        "trouble": false
                                    }
                                },
                                {
                                    "attribute": 17,
                                    "data": "0000000000",
                                    "name": "zone_status1111",
                                    "value": {
                                        "ac_fault": false,
                                        "alarm1": false,
                                        "alarm2": false,
                                        "battery_defect": false,
                                        "low_battery": false,
                                        "restore": false,
                                        "supervision": false,
                                        "tamper": false,
                                        "test_mode": false,
                                        "trouble": false
                                    }
                                }
                            ],
                            "cluster": 1280
                        }
                    ],
                    "device": 1027,
                    "endpoint": 1,
                    "in_clusters": [
                        0,
                        1,
                        3,
                        4,
                        9,
                        1280,
                        1282
                    ],
                    "out_clusters": [
                        3,
                        25
                    ],
                    "profile": 260
                }
            ],
            "generictype": "",
            "info": {
                "addr": "c008",
                "id": 41,
                "ieee": "000d6f00110888eb",
                "last_seen": "2019-04-05 14:15:28",
                "lqi": 132,
                "mac_capability": "10001110",
                "power_type": 1
            }
        },
```

**Comment by ISO-B** [comment]
I have sent test firmware for doudz for testing. I will report back after testing.

**Comment by ISO-B** [comment]
I received my siren. After testing it seems that it supports Squawk from IAS Warning Device Cluster, but I haven't yet managed to successfully use start warning command. For squawk device seems to only support one squawk level.

**Comment by ISO-B** [comment]
Managed to get Start warning command work. I will make PR at Monday or Tuesday.

**Comment by pipiche38** [comment]
But this is already addressed with Firware 3.1a, i have the same and the warning works perfectly.
@akila made it

**Comment by pipiche38** [comment]
```
Zigate -> Obj  0x0111  Write Attribute request IAS_WD (from v3.1a)
                       <address mode: uint8_t>
   <target short address: uint16_t>
   <source endpoint: uint8_t>
   <destination endpoint: uint8_t>
   <direction: uint8_t>
   <manufacturer specific: uint8_t>
   <manufacturer id: uint16_t>
   <Warning Mode: uint8_t>
   <Warning Duration: uint16_t>
   <Strobe duty cycle : uint8_t>
   <Strobe level : uint8_t>

```

**Comment by pipiche38** [comment]
Now, implementing the Squawk command woulld be great too

**Comment by ISO-B** [comment]
Ah, didn't notice that, because I haven't followed progress very closely in summer. I will add squawk.

**Comment by pipiche38** [comment]
Ah ah the finish summer time ;-)

**Comment by pipiche38** [comment]
This can be closed as it is fully addressed in 3.1a
